### PR TITLE
[WEB-1648] Fix Nav Collapse for Guides

### PR DESCRIPTION
**Pull request by DavidWeid**
### What does this PR do?
- When on a subpage of a Guides section, the parent level Guide gets highlighted in the sidnav so the user can keep track of their location
- Updates a couple old links' paths
- Fixes bug with pages containing the string `api/` but not starting with `api/` not having the `js-load` or `active` class applied (url either ending in `-api` or `_api`
- Fix bug where children of "Integrations -> Guides" wouldn't appear in sidenav due to missing class `open` not being applied
- Fix sidnav active display bug with log management and synthetics guides

### Motivation
https://datadoghq.atlassian.net/browse/WEB-1648

### Preview
Navigating through Guide subpages should reflect in the sidenav without populating each guide's name
https://docs-staging.datadoghq.com/davidweid/guides-nav-collapse/events/guides/agent/

### Additional Notes
The following guides need to have their subpages included in the `menus.en.yaml` file:
- integrations
- events
- serverless
- continuous_integration
- security_platform
- real_user_monitoring
- network_monitoring/performance
- network_monitoring/devices

The following guides do not need to have their subpages included in the `menus.en.yaml` file:
- agent
- monitors
- tracing
- logs
- synthetics
- developers

### Categories with Guides
- Agent -> Guides
- Integrations -> Guides
- Events -> Guides
- Dashboards -> Widgets
- Serverless -> Guides
- Monitors -> Guides
- Tracing -> Guides
- Logs -> Guides
- Security Platform -> Guides
- Synthetics -> Guides
- RUM -> Guides
- Network Monitoring -> Guides
- Developers -> Guides

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.

### CODE REVIEW

**Review by stanleegoodspeed — APPROVED** [review]

**Comment by sarina-dd** [comment]
Hi @DavidWeid! Thanks for taking a look into this. Is it possible to have **Guides** highlighted in the left nav, but not populate each individual guide name? We want users to know which section they are in within the nav when looking at a guide, but our guide sections can be long (20+ guides) and we don't want to populate each guide name in the navigation for multiple other purposes.

Also, @nsollecito: `Shows the 4th level of sidenav so some guides can be shown, while manually hiding previously hidden 4th level content` - want to verify if we're 👌 on this. Feels unnecessary if we only populate the level 3 just to give users guidance as to which section their in, so I'm assuming we can remove this portion if so.

**Comment by nsollecito** [comment]
@sarina-dd - agreed, we should stick to 3-levels in the subnav and not expand the Guide sub pages..

**Comment by DavidWeid** [comment]
@sarina-dd Please review the preview and PR again. I've updated this so that the Guide / Section is highlighted in the sidenav without populating the individual guides. Also, 4th level has been reverted to hidden. 🙏🏼

**Comment by sarina-dd** [comment]
@DavidWeid Thanks for the updates! 🔥  LGTM!

File: config/_default/menus/menus.en.yaml [hunk 1]
```diff
@@ -787,6 +787,94 @@ main:
     identifier: agent_guides
     parent: agent
     weight: 11
+  - name: Agent Commands
```
Comment (sarina-dd) [inline config/_default/menus/menus.en.yaml:790]: @DavidWeid Do we still need required these changes to the `menus.en.yaml` file now that we're not listing individually in the nav? I think we can revert, correct?

Comment (DavidWeid) [reply config/_default/menus/menus.en.yaml]: Some of the guide pages need to be listed in the menu for the state to properly display in the sidenav. I reviewed this another time this morning and updated the PR description to note which items are required or not (see additional notes). I've updated the file to remove unneeded subpages while keeping the required ones.

**Review by sarina-dd — APPROVED** [review]

**Pull request merged**